app/models/group.rb

```diff
@@ -91,6 +91,26 @@ def shared_namespace_metadata_keys(namespace)
     metadata_fields
   end
 
+  def aggregated_samples_count # rubocop:disable Metrics/AbcSize
```

Comment:
Nitpick: Please change the order of the first 2 lines of the function to
```
return sample_count unless shared_namespaces.any?

aggregated_samples_count = samples_count
```
This way the first line is the early return from the function, and we do not have to create/assign the new variable before doing so.